include/eosio/vm/allocator.hpp

```diff
@@ -131,26 +130,20 @@ namespace eosio { namespace vm {
       T* alloc(size_t size = 1 /*in pages*/) {
          EOS_WB_ASSERT(page + size <= max_pages, wasm_bad_alloc, "exceeded max number of pages");
          mprotect(raw + (page_size * page), (page_size * size), PROT_READ | PROT_WRITE);
-         T* ptr    = (T*)_previous;
-         _previous = (raw + (page_size * page));
+         T* ptr    = (T*)(raw + (page_size * page));
+         memset(ptr, 0, page_size * size);
```

Comment:
I thought mmap + MAP_ANON zero-initializes the pages.  Is that not the case?

Reply:
The linux kernel has weird performance-enhancing exceptions to zero-initializing. Its idea is that reusing a page from the same process isn't a problem.

Reply:
It actually hit this case. Some pages came back non-0.